driver-core/src/main/com/mongodb/connection/netty/ReadTimeoutHandler.java

```diff
@@ -40,13 +40,16 @@ public ReadTimeoutHandler(final long readTimeout) {
     }
 
     void scheduleTimeout(final ChannelHandlerContext ctx) {
-        timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        if (timeout == null) {
```

Comment:
The default will be `null` for timeout and we can't throw if `timeout != null` because of how the `NettyStream` works.

This can be the flow:

1. `readAsync` called.
     `scheduleReadTimeout()` called.
      if no bytes available - `pendingReader` is set.
      Nothing else todo.

2. Later `InboundBuffer` is called with some bytes
    `handleReadResponse` is called, which in turn will call `readAsync`
    `scheduleReadTimeout()` is called (we already have a timeout set!).
    Check if the number of bytes received is enough - if not set `pendingReader` and eventually repeat this step - or if it is enough complete the handler!